src/main/java/reactor/netty/ReactorNetty.java

```diff
@@ -837,9 +837,30 @@ public synchronized Throwable fillInStackTrace() {
 	static final ConnectionObserver NOOP_LISTENER = (connection, newState) -> {};
 
 	static final Logger log                               = Loggers.getLogger(ReactorNetty.class);
-	static final AttributeKey<Boolean> PERSISTENT_CHANNEL = AttributeKey.newInstance("$PERSISTENT_CHANNEL");
 
-	static final AttributeKey<Connection> CONNECTION = AttributeKey.newInstance("$CONNECTION");
+	static final AttributeKey<Boolean> PERSISTENT_CHANNEL = getOrCreateAttribute("$PERSISTENT_CHANNEL", Boolean.class);
+
+	static final AttributeKey<Connection> CONNECTION = getOrCreateAttribute("$CONNECTION", Connection.class);
+
+	static <T> AttributeKey<T> getOrCreateAttribute(String name, Class<T> attributeClass) {
+		if (AttributeKey.exists(name)) {
```

Comment:
I think that we can safely invoke just `AttributeKey.valueOf(name)`
https://github.com/netty/netty/blob/83895f0f7a784225333c99a194009e7f9b7c7406/common/src/main/java/io/netty/util/AttributeKey.java#L34-L40
https://github.com/netty/netty/blob/83895f0f7a784225333c99a194009e7f9b7c7406/common/src/main/java/io/netty/util/ConstantPool.java#L46-L57

Wdyt?

Reply:
Indeed your are right. I made the change and override my last commit.